src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs

```diff
@@ -26,8 +26,7 @@
 
 namespace Microsoft.ML.Data
 {
-    [BestFriend]
-    internal sealed class RankingEvaluator : EvaluatorBase<RankingEvaluator.Aggregator>
+    public sealed class RankingEvaluator : EvaluatorBase<RankingEvaluator.Aggregator>
```

Comment:
Is exposing the four classes the right method?

/cc @eerhardt

Reply:
No this is not the right approach. @harishsk - can you follow the same pattern we are following for the other evaluators? None of the other evaluator classes are public. There should be no reason why this is.

---
In reply to: [314017216](https://github.com/dotnet/machinelearning/pull/4081#discussion_r314017216) [](ancestors = 314017216)